### PR TITLE
Add CMakeLists.txt

### PR DESCRIPTION
**Pull request by CaseyCarter**
This replaces the horribly bit-rotted Visual Studio projects with a modern CMake build system. The CMake build can't yet do everything the Makefile build does, which is fine: the goal isn't to replace it, but to have a working MSVC build.

This shouldn't be destabilizing: the only interaction with the Makefile build/CI is the bash script to run xmllint, which is well tested. It should be safe to apply during the meeting.

### CODE REVIEW

File: bin/lint.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+find xml -name 'issue*.xml' -print0 | \
+xargs -0 -P $(getconf _NPROCESSORS_ONLN) -n 128 xmllint --noout --nowarning --dtdvalid xml/lwg-issue.dtd
```
Comment (CaseyCarter) [inline bin/lint.sh:3]: This looks horrible to me with the pipeline split onto two lines with the same indentation; is there an idiomatic way to format it?

Comment (jwakely) [reply bin/lint.sh:3]: And if you want shorter lines, you could even break between the xargs options and the command it runs, with another level of indentation:
```sh
find xml -name 'issue*.xml' -print0 \
  | xargs -0 -P $(getconf _NPROCESSORS_ONLN) -n 128 \
    xmllint --noout --nowarning --dtdvalid xml/lwg-issue.dtd
```

**Review by jwakely — APPROVED** [review]
This seems fine as is, but if you want to reformat the lint.sh that's fine too.

**Review by jwakely — APPROVED** [review]

**Review by Dani-Hub — APPROVED** [review]

**Comment by jwakely** [comment]
Rebased with fixups and pushed as d5afd6e5c9b5a8cfc5a08c6ea442990b6bf2a2ea

**Pull request closed without merging**